lib/blocks.php

```diff
@@ -45,3 +45,58 @@ function gutenberg_reregister_core_block_types() {
 	}
 }
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
+
+/**
+ * Registers a new block style.
+ *
+ * @param string $block_name       Block type name including namespace.
+ * @param array  $style_properties Array containing the properties of the style name, label, style (name of the stylesheet to be enqueued), inline_style (string containing the CSS to be added).
+ */
+function gutenberg_register_block_style( $block_name, $style_properties ) {
+	if ( isset( $style_properties[ 'style' ] ) ) {
+		$style_handle = $style_properties[ 'style' ];
+		unset( $style_properties[ 'style' ] );
+		add_action(
+			'enqueue_block_assets',
+			function() use ( $style_handle ) {
```

Comment:
This is probably not compatible with php 5.2. If we keep it, we should update the minimum version of the plugin to 5.2 (which I think could be considered)

Reply:
Probably upgrading minimum plugin version is the way to go. As an aside, I'm probably missing something but with PHP 5.2 the only way to express this would be using a class where we pass the $style_handle handle and then that class has a function that calls `wp_enqueue_style( $this->style_handle );` or using something like create_function which was deprecated. The PHP 5.2 options that come to my mind all seem verbose.

Reply:
PR for WordPress 5.2 minimum (PHP 5.6) #15809